integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,306 @@
+package main_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+	"time"
+
+	"os/exec"
+
+	"lightbend.com/cloudflow/itest/cli"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Assumptions:
+// In this test, we are going to deploy the test application to an existing cluster.
+// For this first step, we assume that the cluster is correctly deployed and that kubectl is configured to talk to it.
+// The application is already built
+// We also assume that the test is going to be executed manually.
+// Automation will be the next step
+
+const (
+	JsonTokenSrc    = "/keybase/team/assassins/gcloud/pipelines-serviceaccount-key-container-registry-read-write.json"
+	ShortTimeout    = 60  // seconds
+	LongTimeout     = 240 // seconds
+	InitialWaitTime = "30s"
+)
+
+var swissKnifeApp = cli.App{
+	Image: "eu.gcr.io/bubbly-observer-178213/swiss-knife:189-277e424",
+	Name:  "swiss-knife",
+}
+
+var _ = Describe("Application deployment", func() {
+	Context("the cluster is clean for testing", func() {
+		It("should not have the test app", func() {
+			err := ensureAppNotDeployed(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("when I deploy an application", func() {
+		It("should start a deployment", func() {
+			jsonToken := getToken()
+			output, err := cli.Deploy(swissKnifeApp, jsonToken)
+			Expect(err).NotTo(HaveOccurred())
+			expected := "Deployment of application `" + swissKnifeApp.Name + "` has started."
+			Expect(output).To(ContainSubstring(expected))
+			waitTime, _ := time.ParseDuration(InitialWaitTime)
+			time.Sleep(waitTime) // this wait is needed to let the application deploy
```

Comment:
I agree with the idea discussed yesterday that watching the cluster for the relevant events instead of waiting a set time will be a better way to determine when the deployment has succeeded or failed.

If this is hard, for instance if we can't just monitor status updated on the application CR, that should lead to feature requests. E.g. we should not be polling all expected pods sine that is an implementation detail of the Cloudflow operator. We should watch for the operator to mark the application (or all its pods) as running in the Status section of the CR.

Reply:
Would a user subscribe to k8s events to know that the deployment succeeded? 
The intention here is to use the tools we offer to the users to exercise the cloudflow application lifecycle. 
In this case, we have:
```
kubectl cloudflow deploy <app>
```
which is followed later by:
```
kubectl cloudflow status <app>
-> has status my expected streamlets?
```

Reply:
Exactly. I was reacting to the explicit wait, which you explained is only there due to a bug in the current implementation of application status. I wold appreciate a comment above this line and a link to the relevant issue so any reader will known why we are doing this and can see when it will be fixed.

Reply:
created a public ticket and added a link to it:
https://github.com/lightbend/cloudflow/issues/201